Assignment5/GenericsHomwork.Tests/NodeTests.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using GenericsHomework;
+using System;
+
+namespace GenericsHomwork.Tests
+{
+    [TestClass]
+    public class NodeTests
+    {
+        [TestMethod]
+        public void Constructor_ValidInput_Success()
+        {
+            int number = 4;
+            Node<int> testNode = new Node<int>(number);
+            Assert.IsNotNull(testNode);
+        }
+        [ExpectedException(typeof(ArgumentNullException))]
+        [TestMethod]
+        public void Constructor_NullInput_Success()
+        {
+            int? number = null;
+            Node<int?> testNode = new Node<int?>(number);

```

Comment:
Because `testNode` is not used you can use a discard. In C# 9, this actually will tell the compiler to not even both to generate a variable for this (previously versions of C# this would still be a variable with the name of "_").
```suggestion
            _ = new Node<int?>(number);
```